### PR TITLE
Feat: new builds table

### LINKED ISSUE #387: Tree Details Page: Sort tables (all tabs) by columns

### PR DESCRIPTION
**Pull request by MarceloRobert**
Converts the old build tab of Tree Details page to use tanstack-table. This allows to sort any desired column.

To test, go to any tree detail with builds data, at builds tab, scroll down and see the new table.

Closes #387

### CODE REVIEW

File: dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTable.tsx [hunk 1]
```diff
@@ -315,51 +315,51 @@ export function TestsTable({ testHistory }: ITestsTable): JSX.Element {
...
-              ))}
+  const tableRows = useMemo((): JSX.Element[] | JSX.Element => {
+    return modelRows?.length ? (
+      modelRows.map(row => (
+        <Fragment key={row.id}>
+          <TableRow
+            className="group cursor-pointer hover:bg-lightBlue"
+            onClick={() => {
+              if (row.getCanExpand()) row.toggleExpanded();
+            }}
```
Comment (lfjnascimento) [inline dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTable.tsx:326]: `useCallback` here

File: dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTable.tsx [hunk 2]
```diff
@@ -242,6 +224,24 @@ export function TestsTable({ testHistory }: ITestsTable): JSX.Element {
       return count;
     }, [rawData]);
 
+  const onClickFilter = useCallback(
```
Comment (lfjnascimento) [inline dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTable.tsx:227]: Any reason to move this block of code?

**Review by lfjnascimento — APPROVED** [review]
LGTM

File: dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTable.tsx (hunk 2 above)
Comment (MarceloRobert) [reply dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTable.tsx:227]: Makes the order of the functions/variables consistent with the other tables. rawData - data - filterCount - onClickFilter - filters - table - headers - rows - return

**Review by Francisco2002 — APPROVED** [review]
It is working nice!!

**Pull request merged**